File: leetcode/lc-us/cpp/27.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int removeElement(vector<int> &nums, int val) {
        int low = 0;
        for (int i = 0; i < nums.size(); i++) {
            while (low < nums.size() && nums[low] != val) {
                low++;
            }
            if (nums[i] != val && i > low) {
                swap(nums[low++], nums[i]);
            }
        }
        return low;
    }
};

File: leetcode/lc-us/rust/27.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut i: i32 = 0;
        for j in 0..nums.len() {
            if nums[j] == val {
                continue;
            }
            nums[i as usize] = nums[j];
            i += 1;
        }
        i
    }
}